gap/prop.gi

```diff
@@ -8,7 +8,47 @@
 #############################################################################
 ##
 
-#
+InstallMethod(IsMeetSemilatticeDigraph, "for a digraph",
+[IsDigraph],
+function(digraph)
+  local out_nbrs, i, j, intr;
+  if IsPartialOrderDigraph(digraph) then
+    out_nbrs := OutNeighbours(digraph);
+    for i in [1 .. Size(out_nbrs)] do
+      for j in [i + 1 .. Size(out_nbrs)] do
+        # Is there a better way of doing this? JDM
+        intr := Intersection(out_nbrs[i], out_nbrs[j]);
+        if not intr in out_nbrs then
+          return false;
+        fi;
+      od;
+    od;
+    return true;
+  else
+    return false;
+  fi;
+end);
+
+InstallMethod(IsJoinSemilatticeDigraph, "for a digraph",
+[IsDigraph],
+function(digraph)
+  local in_nbrs, intr, i, j;
+  if IsPartialOrderDigraph(digraph) then
+    in_nbrs := InNeighbours(digraph);
+    for i in [1 .. Size(in_nbrs)] do
+      for j in [i + 1 .. Size(in_nbrs)] do
+        # Is there a better way of doing this? JDM
+        intr := Intersection(in_nbrs[i], in_nbrs[j]);
+        if not intr in in_nbrs then
+          return false;
+        fi;
+      od;
+    od;
```

Comment:
Unless I'm mistaken, the function `IsJoinSemilatticeDigraph(digraph)` does exactly the same as `IsMeetSemilatticeDigraph(DigraphReverse(digraph))`. Is this mathematically the right thing to do? I think so; in which case you should replace this whole function which one that just calls `IsMeetSemilatticeDigraph` on the reversed digraph.